### PR TITLE
nxos_snmp_community: Use running config for module operations instead of cli commands. 

### PR DESCRIPTION
**Pull request by GGabriele**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
nxos_snmp_community

##### ANSIBLE VERSION
```
ansible 2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY
Changed how module work to make it use running config for module operations instead of cli commands. Referring to #5214

### CODE REVIEW

**Comment by GGabriele** [comment]
Migrated from https://github.com/ansible/ansible-modules-core/pull/5586

**Comment by gundalow** [comment]
@GGabriele rebase needed

File: lib/ansible/modules/network/nxos/nxos_snmp_community.py [hunk 1]
```diff
@@ -58,6 +58,29 @@
...
         default: present
         choices: ['present','absent']
+    include_defaults:
+        description:
+            - Specify to use or not the complete runnning configuration
+              for module operations.
+        required: false
+        default: false
+        choices: ['true','false']
+        version_added: 2.3
```
Comment (gundalow) [inline lib/ansible/modules/network/nxos/nxos_snmp_community.py:68]: `"2.4"`

File: lib/ansible/modules/network/nxos/nxos_snmp_community.py [hunk 2]
```diff
@@ -58,6 +58,29 @@
...
+    config:
+        description:
+            - Configuration string to be used for module operations. If not
+              specified, the module will use the current running configuration.
+        required: false
+        default: null
+        version_added: 2.3
```
Comment (gundalow) [inline lib/ansible/modules/network/nxos/nxos_snmp_community.py:75]: `"2.4"`

File: lib/ansible/modules/network/nxos/nxos_snmp_community.py [hunk 3]
```diff
@@ -58,6 +58,29 @@
...
+    save:
+        description:
+            - Specify to save the running configuration after
+              module operations.
+        required: false
+        default: false
+        choices: ['true','false']
+        version_added: 2.3
```
Comment (gundalow) [inline lib/ansible/modules/network/nxos/nxos_snmp_community.py:83]: `"2.4"`

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by calfonso** [comment]
@GGabriele just checking in on this PR, are you going to apply the required changes to get this merged?

**Comment by gundalow** [comment]
needs_rebase

**Comment by gundalow** [comment]
@GGabriele Hi, are you able to update this?

**Comment by trishnaguha** [comment]
@GGabriele `include_defaults, config, save` options are deprecated since 2.4.0. There have been introduced modifications to the module. Please pull the devel branch and give it a try. Closing the PR since there is no response for long time, feel free to reopen/ open new PR if the issue is not fixed. 
Thanks!

**Pull request closed without merging**